controllers/sr_controller/sr_controller.py

```diff
@@ -134,14 +135,24 @@ def main():
     env['SR_ROBOT_MODE'] = robot_mode
     env['SR_ROBOT_FILE'] = str(robot_file)
 
-    completed_process = subprocess.run(
+    child_process = subprocess.Popen(
         [sys.executable, "-u", str(robot_file)],
         env=env,
         cwd=str(robot_file.parent),
     )
 
+    def terminate_handler(*args, **kwargs):
+        child_process.terminate()
+        sys.exit(0)
+
+    # Ensure the child dies
+    signal.signal(signal.SIGKILL, terminate_handler)
+    signal.signal(signal.SIGTERM, terminate_handler)
```

Comment:
This line doesn't seem to work on Windows. However, after removing it the processes still seem to stick around when resetting webots.

Reply:
Might be worth looking into [PSUtil](https://pypi.org/project/psutil/) for this.

If we don't want to add another dependency, it is BSD licensed, so we can just select the right parts of that library to include in our code.